projects/ngx-form-object/src/lib/helpers/update-model-property-definition/update-model-property-definition.helper.ts

```diff
@@ -0,0 +1,23 @@
+import { PropertyOptions } from '../../interfaces/property-options.interface';
+
+export function updateModelPropertyDefinition(
+	target: Record<string, unknown>,
+	propertyName: string | symbol,
+	internalModelProperty: symbol,
+	options: PropertyOptions
+): void {
+	const modelAttributePropertiesDescriptor: PropertyDescriptor = Object.getOwnPropertyDescriptor(
+		target,
+		internalModelProperty
+	) || { value: new Map() };
+	const attributeProperties: Map<string | symbol, PropertyOptions> = modelAttributePropertiesDescriptor.value;
+
+	attributeProperties.set(propertyName, options);
```

Comment:
`attributeProperties` seems a bit confusing as this function is also called for BelongsTo and HasMany relationships.